services/client.go

```diff
@@ -0,0 +1,135 @@
+package services
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/frain-dev/convoy/datastore"
+	"github.com/gorilla/websocket"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// Maximum message size allowed from peer.
+	maxMessageSize = 512
+
+	maxDeviceLastSeenDuration = 2 * time.Minute
+)
+
+// Client is a middleman between the websocket connection and the hub.
+type Client struct {
+	hub *Hub
+
+	// device id of the cli client
+	deviceID   string
+	EventTypes []string
```

Comment:
We create subscriptions off the client's request. Is there any reason why we need `EventTypes` on this type?